File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int size = strs.size()-1;
        int swrd = strs[0].size()-1;
        string res = "";
        sort(strs.begin(),strs.end());
        int n = strs.size();
        string first=strs[0],last=strs[n-1];
        for(int j = 0;j < min(first.size(),last.size()) ; j++){
                 if(first[j]!=last[j]){
                return res;
            }
            res += strs[0].at(j);
        }
        return res;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1: return strs[0]
        small = strs[0]
        for i in strs[1:]:
            if len(i) < len(small):
                small = i 
        
        # print(small)
        for i in strs: 
            while not i.startswith(small):
                small = small[:-1]
                if not small: return ""
        return small
             

        
    
            
        